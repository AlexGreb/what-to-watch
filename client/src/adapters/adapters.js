export const createMovie = (movie) => {
  return {
    title: movie.name,
    genre: movie.genre,
    date: movie.released,
    backgroundColor: movie.backgroundColor,
    background: movie.backgroundImage,
    poster: movie.previewImage,
    posterImage: movie.posterImage,
    id: movie.id,
    description: movie.description,
    rating: movie.rating,
    votes: movie.scoresCount,
    director: movie.director,
    starring: movie.starring,
    runTime: movie.runTime,
    preview: movie.previewVideoLink,
    videoLink: movie.videoLink,
    isFavorite: movie.isFavorite,
  };
};

export const createReview = (review) => {
  return {
    id: review.id,
    name: review.user.name,
    userId: review.user.id,
    rating: review.rating,
    date: review.date,
    comment: review.comment
  };
};

const Rating = {
  AWESOME: 10,
  VERY_GOOD: 8,
  GOOD: 5,
  NORMAL: 3,
};

export const getRatingValue = (rating) => {
  if (rating === Rating.AWESOME) {
    return `Awesome`;
  }
  if (rating < Rating.AWESOME && rating >= Rating.VERY_GOOD) {
    return `Very good`;
  }
  if (rating < Rating.VERY_GOOD && rating >= Rating.GOOD) {
    return `Good`;
  }
  if (rating < Rating.GOOD && rating >= Rating.NORMAL) {
    return `Normal`;
  }
  return `Bad`;
};
