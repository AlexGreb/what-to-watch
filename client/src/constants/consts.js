export const tabs = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const buttonTypes = {
  SUBMIT: `submit`,
  RESET: `reset`,
  BUTTON: `button`
};

export const VIDEO_PROGRESS = 100;

export const MOVIES_TO_SHOW = 8;

export const BASE_URL = `https://15.react.htmlacademy.pro/wtw/`;

// export const BASE_URL = `http://localhost:3002/api/v1/`;

export const API_TIMEOUT = 5000;

export const Errors = {
  UNAUTHORIZED: 401
};
//  TODO брать из сервера
export const dataUrl = {
  HOME: `/`,
  FILMS: `/films/`,
  PROMO: `/promo`,
  LOGIN: `/login`,
  FAVORITE: `/favorite/`,
  PLAYER: `/player/`,
  REVIEWS: `/comments/`,
  // FILMS: `/movies`,
  // PROMO: `/promo`,
  // LOGIN: `/login`,
  // FAVORITE: `/favorite/`,
  // PLAYER: `/player/`,
  // REVIEWS: `/comments/`,
};


export const authorizationStatus = {
  AUTH: `AUTH`,
  NO_AUTH: `NO_AUTH`
};
