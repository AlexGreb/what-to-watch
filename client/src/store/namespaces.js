export const namespaces = {
  APP: `app`,
  MOVIES: `movies`,
  USER: `user`
};
