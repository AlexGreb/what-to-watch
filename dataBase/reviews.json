[
  {
    "movieId": 1,
    "reviews": [{"id":1,"user":{"id":17,"name":"Emely"},"rating":6.5,"comment":"This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.","date":"2024-01-07T07:19:48.519Z"}]
  },
  {
    "movieId": 2,
    "reviews":[{"id":1,"user":{"id":12,"name":"Isaac"},"rating":6.9,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-14T07:19:48.519Z"}]
  },
  {
    "movieId": 3,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":9,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-08T07:19:48.519Z"}]
  },
  {
    "movieId": 4,
    "reviews":[]
  },
  {
    "movieId": 5,
    "reviews":[{"id":1,"user":{"id":16,"name":"Mollie"},"rating":5.7,"comment":"This movie really touched my heart, it really is the best movie of the year and everyone should see this masterpiece.","date":"2024-01-19T07:19:48.519Z"}]
  },
  {
    "movieId": 6,
    "reviews":[]
  },
  {
    "movieId": 7,
    "reviews":[{"id":1,"user":{"id":10,"name":"Max"},"rating":1.2,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-07T07:19:48.519Z"}]
  },
  {
    "movieId": 8,
    "reviews":[]
  },
  {
    "movieId": 9,
    "reviews":[]
  },
  {
    "movieId": 10,
    "reviews":[]
  },
  {
    "movieId": 11,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":5.4,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-22T07:19:48.519Z"}]
  },
  {
    "movieId": 12,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":3.8,"comment":"I really find it difficult to believe this movie is rated highly by people. It's hands down the worst movie I've seen in my life","date":"2024-01-24T07:19:48.519Z"}]
  },
  {
    "movieId": 13,
    "reviews":[{"id":1,"user":{"id":12,"name":"Isaac"},"rating":9.2,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-09T07:19:48.519Z"}]
  },
  {
    "movieId": 14,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":6.8,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-05T07:19:48.519Z"},{"id":2,"user":{"id":16,"name":"Mollie"},"rating":7.2,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-26T07:19:48.519Z"}]
  },
  {
    "movieId": 15,
    "reviews":[{"id":1,"user":{"id":18,"name":"Sophie"},"rating":1.1,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 16,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":2.4,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-11T07:19:48.519Z"},{"id":2,"user":{"id":14,"name":"Corey"},"rating":4.9,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-22T07:19:48.519Z"},{"id":3,"user":{"id":17,"name":"Emely"},"rating":3.4,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-30T07:19:48.519Z"}]
  },
  {
    "movieId": 17,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":5.6,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-10T07:19:48.519Z"},{"id":2,"user":{"id":17,"name":"Emely"},"rating":6.5,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-06T07:19:48.519Z"}]
  },
  {
    "movieId": 18,
    "reviews":[{"id":1,"user":{"id":15,"name":"Kendall"},"rating":2.6,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-08T07:19:48.519Z"},{"id":2,"user":{"id":14,"name":"Corey"},"rating":4.1,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-13T07:19:48.519Z"}]
  },
  {
    "movieId": 19,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":4.9,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-01-11T07:19:48.519Z"}]
  },
  {
    "movieId": 20,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":8.1,"comment":"Unfortunately we don't have a reliable way to tell the true ratings of a movie.","date":"2024-02-01T07:19:48.519Z"},{"id":2,"user":{"id":11,"name":"Jack"},"rating":6.4,"comment":"It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 21,
    "reviews":[{"id":1,"user":{"id":11,"name":"Jack"},"rating":6.3,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-26T07:19:48.519Z"},{"id":2,"user":{"id":12,"name":"Isaac"},"rating":1.5,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-31T07:19:48.519Z"},{"id":3,"user":{"id":10,"name":"Max"},"rating":2.6,"comment":"I personally found this movie to be boring. Definitely one of the most boring movies I've ever seen.","date":"2024-01-30T07:19:48.519Z"}]
  },
  {
    "movieId": 22,
    "reviews":[{"id":1,"user":{"id":10,"name":"Max"},"rating":8.5,"comment":"I love this movie. This film is a milestone in cinematography. Great Immersive camera-work. This film is an experience and i has already seen it 4 times and I only see more quality of the film.","date":"2024-01-20T07:19:48.519Z"},{"id":2,"user":{"id":16,"name":"Mollie"},"rating":7.1,"comment":"Unfortunately we don't have a reliable way to tell the true ratings of a movie.","date":"2024-01-31T07:19:48.519Z"},{"id":3,"user":{"id":16,"name":"Mollie"},"rating":4.2,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-21T07:19:48.519Z"}]
  },
  {
    "movieId": 23,
    "reviews":[{"id":1,"user":{"id":16,"name":"Mollie"},"rating":4.5,"comment":"It was well acted, directed, and the music was good. But the story is yawn. Not trying to rip anybody but I checked my watch a dozen times during this movie.","date":"2024-02-01T07:19:48.519Z"},{"id":2,"user":{"id":13,"name":"Zak"},"rating":1.6,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-16T07:19:48.519Z"},{"id":3,"user":{"id":14,"name":"Corey"},"rating":2,"comment":"Poised to be an instant classic, almost everything about this film is phenomenal - the acting, the cinematography, the discography, etc.","date":"2024-01-18T07:19:48.519Z"}]
  },
  {
    "movieId": 24,
    "reviews":[{"id":1,"user":{"id":19,"name":"Christina"},"rating":9.4,"comment":"This movie is just plain bad. There must be some big payola going round this awards season. Badly written, average acting at best, all the characters are unrelatable and inlikeable. 2 hours of my life wasted.","date":"2024-02-01T07:19:48.519Z"}]
  },
  {
    "movieId": 25,
    "reviews":[{"id":1,"user":{"id":14,"name":"Corey"},"rating":3.7,"comment":"The editing is a mess, and the transitions of the plot or characters are rather strange. There is no narrative focus and the storytelling is unbalanced. I cannot really understand why such a bad movie received an overwhelming approval from the critics. ","date":"2024-01-18T07:19:48.519Z"},{"id":2,"user":{"id":11,"name":"Jack"},"rating":6.2,"comment":"A movie that will take you to another world full of emotions.","date":"2024-01-10T07:19:48.519Z"}]
  }
]

