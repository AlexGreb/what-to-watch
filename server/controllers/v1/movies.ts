import type {ResponseData} from "../../types";
import type {IMovie} from "../../Models/movie";
import type {Request, Response} from "express";
import * as movieService from "../../services/v1/movieService";
import {getFavoriteMovies} from "../../services/v1/favoriteService";
import {getResError, setFavoriteMovies} from "../../utils"
import {validationResult} from "express-validator";

export const getAllMovies = async (req: Request, res: Response): Promise<void> => {
    try {
        const errors = validationResult(req);
        if(!errors.isEmpty()) {
            res.status(404).json({
                status: 'FAILED',
                data: {
                    error: errors,
                }
            })
        }
        const movies: IMovie[] = await movieService.getAllMovies();
        const favoriteMovies = await getFavoriteMovies(Number(req.params.userId));
        const moviesWithFavoriteUser = setFavoriteMovies(favoriteMovies, movies);
        const data: ResponseData<IMovie[]> = {
            status: 'OK',
            data: moviesWithFavoriteUser
        }
        res.json(data);
    } catch(error: unknown) {
        const {status, data} = getResError(error);
        res.status(status).json({
            status: 'FAILED',
            data
        });
    }
}

export const getMovie = async (req: Request, res: Response): Promise<void> => {
    const {params} = req;
    if(!params.movieId) return;
    try {
        const movie: IMovie | null = await movieService.getMovie(Number(params.movieId));

        if(movie) {
            res.status(200).json({
                status: 'OK',
                data: movie,
            })
        } else {
            res.status(404).json({
                status: 'FAILED',
                data: {
                    error: "Movie with this movieId wasn`t found",
                }
            })
        }
    } catch(error) {
        const {status, data} = getResError(error);
        res.status(status).json({
            status: 'FAILED',
            data
        });
    }
}

export const getPromo = async (req: Request, res: Response): Promise<void> =>{
    try {
        const promo: IMovie | null = await movieService.getPromo();
        res.json({
            status: 'OK',
            data: promo
        });
    } catch(error) {
        const {status, data} = getResError(error);
        res.status(status).json({
            status: 'FAILED',
            data
        });
    }
}
