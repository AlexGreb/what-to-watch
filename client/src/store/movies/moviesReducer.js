import {createSlice, createAsyncThunk} from '@reduxjs/toolkit';
import {createMovie, createReview} from '../../adapters/adapters';
import {dataUrl} from '../../constants/consts.js';
import {namespaces} from '../namespaces';

const initialState = {
  promoMovie: {},
  loadingMovies: false,
  loadingMyMovies: false,
  reviews: [],
  sendingReviewStatus: false,
  myListMovies: [],
  moviesList: [],
  isError: false
};

const Operations = {
  fetchMoviesList: createAsyncThunk(
      `${namespaces.MOVIES}/fetchMoviesList`,
      async (_, {extra}) => {
        const {data} = await extra.get(dataUrl.FILMS);
        return data.data.map(createMovie);
      }),
  fetchPromoMovie: createAsyncThunk(
      `${namespaces.MOVIES}/promo`,
      async (_, {extra}) => {
        const response = await extra.get(dataUrl.PROMO);
        return createMovie(response.data.data);
      }),
  fetchReviews: createAsyncThunk(
      `${namespaces.MOVIES}/fetchReviews`,
      async (movieId, {extra}) => {
        const response = await extra.get(`${dataUrl.REVIEWS}${movieId}`);
        return response.data.data.map(createReview);
      }
  ),
  fetchComment: createAsyncThunk(
      `${namespaces.MOVIES}/fetchComment`,
      async (reviewData, {extra}) => {
        const response = await extra.post(`${dataUrl.REVIEWS}${reviewData.movieId}`, reviewData.review);
        return response.data;
      }
  ),
  fetchMyListMovies: createAsyncThunk(
      `${namespaces.MOVIES}/fetchMyListMovies`,
      async (_, {extra}) => {
        const response = await extra.get(`${dataUrl.FAVORITE}`);
        return response.data.map(createMovie);
      }
  ),
  fetchChangeFavoriteStateMovie: createAsyncThunk(
      `${namespaces.MOVIES}/fetchChangeFavoriteStateMovie`,
      async (movie, {dispatch, fulfillWithValue, extra}) => {
        const response = await extra.post(`${dataUrl.FAVORITE}${movie.id}/${movie.isFavorite ? 0 : 1}`);
        dispatch(Operations.fetchMoviesList());
        dispatch(Operations.fetchPromoMovie());
        return fulfillWithValue(response.data);
      }
  )
};

const moviesSlice = createSlice({
  name: namespaces.MOVIES,
  initialState,
  reducers: {
    setSendingReviewStatus: (state, action) => {
      state.sendingReviewStatus = action.payload;
    },
  },
  extraReducers: {
    [Operations.fetchMoviesList.fulfilled]: (state, action) => {
      state.loadingMovies = false;
      state.moviesList = action.payload;
    },
    [Operations.fetchMoviesList.pending]: (state) => {
      state.loadingMovies = true;
    },
    [Operations.fetchMoviesList.rejected]: (state) => {
      state.isError = true;
    },
    [Operations.fetchPromoMovie.fulfilled]: (state, action) => {
      state.promoMovie = action.payload;
    },
    [Operations.fetchPromoMovie.rejected]: (state) => {
      state.isError = true;
    },
    [Operations.fetchReviews.fulfilled]: (state, action) => {
      state.reviews = action.payload;
    },
    [Operations.fetchComment.fulfilled]: (state) => {
      state.sendingReviewStatus = true;
    },
    [Operations.fetchComment.rejected]: (state) => {
      state.sendingReviewStatus = false;
    },
    [Operations.fetchMyListMovies.fulfilled]: (state, action) => {
      state.loadingMyMovies = false;
      state.myListMovies = action.payload;
    },
    [Operations.fetchMyListMovies.pending]: (state) => {
      state.loadingMyMovies = true;
    },
    [Operations.fetchMyListMovies.rejected]: (state) => {
      state.loadingMyMovies = false;
      state.isError = true;
    }
  }
});

export const {setSendingReviewStatus} = moviesSlice.actions;
export {
  Operations,
  initialState
};
export default moviesSlice.reducer;

