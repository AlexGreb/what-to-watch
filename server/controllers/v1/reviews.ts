import {Request, Response} from "express";
import type {MovieReviews} from "../../Models/reviews";
import * as reviewsService from "../../services/v1/reviewsService";
import {getResError} from "../../utils";

export const getMovieReviews = async (req: Request, res: Response): Promise<void> => {
    const {params} = req;
    if(!params.movieId) return;
    try {
        const reviews: MovieReviews | null = await reviewsService.getMovieReviews(Number(params.movieId));
        res.json({
            status: 'OK',
            data: reviews
        });
    } catch (error) {
        const {status, data} = getResError(error);
        res.status(status).json({
            status: 'FAILED',
            data
        });
    }
}
