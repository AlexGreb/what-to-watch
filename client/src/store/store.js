import {configureStore} from '@reduxjs/toolkit';
import appReducer from './app/appReducer';
import moviesReducer from './movies/moviesReducer.js';
import userReducer from './user/userReducer.js';
import {namespaces} from './namespaces';
import {createApi} from '../api.js';


const api = createApi((...args) => store.dispatch(...args));

export const store = configureStore({
  reducer: {
    [namespaces.APP]: appReducer,
    [namespaces.MOVIES]: moviesReducer,
    [namespaces.USER]: userReducer
  },
  middleware: (getDefaultMiddleware) => {
    return getDefaultMiddleware({
      thunk: {
        extraArgument: api
      },
    });
  },
  devTools: true
  // eslint-disable-next-line no-undef
  // devTools: process.env.NODE_ENV !== `production`,
});

// export type State = ReturnType<typeof store.getState>
