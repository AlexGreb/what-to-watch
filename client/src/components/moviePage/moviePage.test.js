/**
 * @jest-environment jsdom
 */
import React from 'react';
import {render} from '@testing-library/react';
import MoviePage from './moviePage.jsx';
import {movie, moviesList} from '../../mocks/movies.js';
import {reviews} from '../../mocks/reviews.js';
import {createMovie} from '../../adapters/adapters.js';
import {tabs} from '../../constants/consts.js';
import {Provider} from 'react-redux';
import {store} from '../../store/store.js';


describe(`moviePage`, () => {
  window.HTMLMediaElement.prototype.load = () => {};
  window.HTMLMediaElement.prototype.play = () => {};
  window.HTMLMediaElement.prototype.pause = () => {};

  it(`Should render correctly a moviePage`, () => {
    const {asFragment} = render(<Provider store={store}><MoviePage movie={createMovie(movie)}
      movies={moviesList}
      reviews={reviews}
      activeTab={tabs.OVERVIEW}
      onChangeActiveTab={jest.fn()}
    /></Provider>);
    expect(asFragment()).toMatchSnapshot();
  });
});
