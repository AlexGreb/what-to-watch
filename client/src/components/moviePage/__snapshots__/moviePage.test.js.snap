// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`moviePage Should render correctly a moviePage 1`] = `
<DocumentFragment>
  <section
    class="movie-card movie-card--full"
  >
    <div
      class="movie-card__hero"
    >
      <div
        class="movie-card__bg"
      >
        <img
          src="https://es31-server.appspot.com/wtw/static/film/preview/pulp-fiction.jpg"
        />
      </div>
      <h1
        class="visually-hidden"
      >
        WTW
      </h1>
      <header
        class="page-header movie-card__head"
      >
        <div
          class="logo"
        >
          <a
            class="logo__link"
          >
            <span
              class="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              class="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              class="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <div
          class="user-block"
        >
          <div
            class="user-block"
          >
            <a
              class="user-block__link"
              href="sign-in.html"
            >
              Sign in
            </a>
          </div>
        </div>
      </header>
      <div
        class="movie-card__wrap"
      >
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          />
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Crime
            </span>
            <span
              class="movie-card__year"
            >
              1994
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
            <a
              class="btn movie-card__button"
              href="add-review.html"
            >
              Add review
            </a>
          </div>
        </div>
      </div>
    </div>
    <div
      class="movie-card__wrap movie-card__translate-top"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster movie-card__poster--big"
        >
          <img
            alt="The Grand Budapest Hotel poster"
            height="327"
            src="https://es31-server.appspot.com/wtw/static/film/poster/Pulp_Fiction.jpg"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <nav
            class="movie-nav movie-card__nav"
          >
            <ul
              class="movie-nav__list"
            >
              <li
                class="movie-nav__item movie-nav__item--active"
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Overview
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Details
                </a>
              </li>
              <li
                class="movie-nav__item"
              >
                <a
                  class="movie-nav__link"
                  href="#"
                >
                  Reviews
                </a>
              </li>
            </ul>
          </nav>
          <div
            class="movie-rating"
          >
            <div
              class="movie-rating__score"
            >
              4.5
            </div>
            <p
              class="movie-rating__meta"
            >
              <span
                class="movie-rating__level"
              >
                Normal
              </span>
              <span
                class="movie-rating__count"
              >
                1635992 ratings
              </span>
            </p>
          </div>
          <div
            class="movie-card__text"
          >
            The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.
            <p
              class="movie-card__director"
            >
              <strong>
                Director: Quentin Tarantino
              </strong>
            </p>
            <p
              class="movie-card__starring"
            >
              <strong>
                Starring: John Travolta, Uma Thurman, Samuel L. Jackson
              </strong>
            </p>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog catalog--like-this"
    >
      <h2
        class="catalog__title"
      >
        More like this
      </h2>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
        >
          <video
            class="player__video"
            height="175"
            poster="undefined"
            preload="none"
            src="undefined"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="movie-page.html"
            />
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <video
            class="player__video"
            height="175"
            poster="undefined"
            preload="none"
            src="undefined"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="movie-page.html"
            />
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <video
            class="player__video"
            height="175"
            poster="undefined"
            preload="none"
            src="undefined"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="movie-page.html"
            />
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <video
            class="player__video"
            height="175"
            poster="undefined"
            preload="none"
            src="undefined"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="movie-page.html"
            />
          </h3>
        </article>
        <article
          class="small-movie-card catalog__movies-card"
        >
          <video
            class="player__video"
            height="175"
            poster="undefined"
            preload="none"
            src="undefined"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="movie-page.html"
            />
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</DocumentFragment>
`;
